api/src/main/java/com/codeforcommunity/rest/ApiRouter.java

```diff
@@ -311,13 +311,21 @@ private void handleSignUp(RoutingContext ctx) {
       response.setStatusCode(400).end();
     }
 
-    boolean success = false;
+    Optional<UserReturn> ret = Optional.empty();
     if (email != null && firstName != null && lastName != null && encryptedPassword != null && currentYear != -1
         && major != null)
-      success = processor.addUser(email, firstName, lastName, encryptedPassword, currentYear, major);
-    if (success)
-      response.setStatusCode(201).end();
-    else {
+      ret = processor.addUser(email, firstName, lastName, encryptedPassword, currentYear, major);
+    String json = "";
+    try {
+      if (ret.isPresent())
+        json = JacksonMapper.getMapper().writeValueAsString(ret.get());
+    } catch (JsonProcessingException e) {
+      e.printStackTrace();
```

Comment:
You probably want to throw an actual exception here. If you have a ret value and you can't parse it into JSON then something is wrong with the code and it's better to fail fast rather than ignore it